### PR TITLE
Enhance error logging for better root cause identification

### PR DESCRIPTION
**Pull request by jcarranzan**
This PR try to improve the errors/exceptions handling and logging to provide more detailed insights when exceptions occur during the execution of the utility.

### CODE REVIEW

File: artifact-version-diff/src/main/java/io/quarkus/qe/PrepareOperation.java [hunk 1]
```diff
@@ -58,13 +61,24 @@ public static void executeProcess(List<String> command, String errorMsg, Path pa
...
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()))) {
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    output.append(line).append("\n");
+                }
+            }
             process.waitFor();
+            if(process.exitValue() != 0) {
+                throw new RuntimeOperationsException(
+                        new RuntimeException("Process exited with code: " + process.exitValue()),
```
Comment (fedinskiy) [inline artifact-version-diff/src/main/java/io/quarkus/qe/PrepareOperation.java]: Why this double wrapping here and below?

Comment (jcarranzan) [reply artifact-version-diff/src/main/java/io/quarkus/qe/PrepareOperation.java]: I did that way, according to this https://docs.oracle.com/en/java/javase/21/docs/api/java.management/javax/management/RuntimeOperationsException.html#%3Cinit%3E(java.lang.RuntimeException,java.lang.String) .

Comment (rsvoboda) [reply artifact-version-diff/src/main/java/io/quarkus/qe/PrepareOperation.java]: That's valid only  when you catch an existing exception like in the case I mentioned it earlier today - see https://github.com/quarkus-qe/quarkus-utilities/pull/13/files#r1785783139 ... you were in ` } catch (IOException | InterruptedException e) { ...`

File: artifact-version-diff/src/main/java/io/quarkus/qe/PrepareOperation.java [hunk 2]
```diff
@@ -58,13 +61,24 @@ public static void executeProcess(List<String> command, String errorMsg, Path pa
...
+                        errorMsg + " Output: " + output
+                );
+            }
             assertEquals(0, process.exitValue(), errorMsg);
         } catch (IOException | InterruptedException e) {
-            throw new RuntimeException(e);
+            throw new RuntimeOperationsException(new RuntimeException(e), errorMsg);
```
Comment (rsvoboda) [inline artifact-version-diff/src/main/java/io/quarkus/qe/PrepareOperation.java]:
```suggestion
            throw new RuntimeOperationsException(e, errorMsg);
```

Comment (jcarranzan) [reply artifact-version-diff/src/main/java/io/quarkus/qe/PrepareOperation.java]: That's not possible in this case, the arguments required for RuntimeOperationsException are :
`RuntimeOperationsException(java.lang.RuntimeException e, String message) `

Comment (rsvoboda) [reply artifact-version-diff/src/main/java/io/quarkus/qe/PrepareOperation.java]: So throw https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/RuntimeException.html#%3Cinit%3E(java.lang.String,java.lang.Throwable)

File: artifact-version-diff/src/main/java/io/quarkus/qe/PrepareOperation.java [hunk 3]
```diff
@@ -58,13 +61,21 @@ public static void executeProcess(List<String> command, String errorMsg, Path pa
...
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    output.append(line).append("\n");
+                }
+            }
             process.waitFor();
+            if(process.exitValue() != 0) {
+                throw new RuntimeException(errorMsg + " Process exited with code: " + process.exitValue() + ". Output: " + output);
+            }
             assertEquals(0, process.exitValue(), errorMsg);
```
Comment (jedla97) [inline artifact-version-diff/src/main/java/io/quarkus/qe/PrepareOperation.java]: Shouldn't this be removed as you checknig it above, so it redundant?

File: artifact-version-diff/src/main/java/io/quarkus/qe/PrepareOperation.java [hunk 4]
```diff
@@ -4,8 +4,11 @@
 import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
 import org.apache.commons.lang3.RandomStringUtils;
 
+import javax.management.RuntimeOperationsException;
```
Comment (jedla97) [inline artifact-version-diff/src/main/java/io/quarkus/qe/PrepareOperation.java]: Can you remove this unused output. At least i don't see it used anymore.

**Review by jedla97 — COMMENTED** [review]
Looked at this and have two comments.

**Review by fedinskiy — APPROVED** [review]

File: artifact-version-diff/src/main/java/io/quarkus/qe/PrepareOperation.java (hunk 3 above)
Comment (jedla97) [reply artifact-version-diff/src/main/java/io/quarkus/qe/PrepareOperation.java]: @jcarranzan thanks for update but atm you don't use process `output` anywhere.  So you just colect it but not print it anywhere. I don't mind have it in if statement as you had before or update assert message. It's up to you what you will choose

**Review by jedla97 — CHANGES_REQUESTED** [review]
As you removed that if statement the captured process output is not used anywhere!

**Review by jedla97 — APPROVED** [review]
Thanks for update. Merging!

**Pull request merged**